pkg/grpc/requestinfo/headers_multimap.go

```diff
@@ -0,0 +1,19 @@
+package requestinfo
+
+import (
+	"net/http"
+)
+
+// HeadersMultiMap is an interface for http.Header and for metadata.MD types.
+// The former will require a wrapper to implement this interface.
+type HeadersMultiMap interface {
```

Comment:
nit: on the naming, maybe requestHeaders might be a better fitting name? I'm not sure I really get the `multi map` part. Request would make it more fitting (including if we move it to the phonehome package) with its usage.

Reply:
`Multi` has been suggested by @rukletsov, as it reflects the fact of multiple values per key. This is, by the way, how `http.Header` interface is different from `metadata.MD`.

Reply:
I still don't really get the multi part since I don't feel it important enough to call it out explicitly, but the opinion isn't strong enough to blick approval.

Reply:
I guess no-one prevents the use of `http.Header` or metadata for response headers as well. Technically, this can also be used for trailers but even best libraries often just call them headers.

The type here is nothing more than a wrapper over `map[string][]string` which is technically a [multimap](https://en.wikipedia.org/wiki/Multimap). The reason I didn't suggest to call it `StringMultimap` and put it somewhere under `maputil` is because I thought Michaël would have non-trivial code related for `http.Header` manipulation here as well so the intent—header manipulation—affects the implementation of the interface.

Reply:
Behold! I removed the interface definition in favor of in-place ones. This also allowed me to move the file to the `telemetry/phonehome` package.

Reply:
Ah, well, not so fast.